pkg/fanal/types/artifact.go

```diff
@@ -62,19 +65,20 @@ type Layer struct {
 }
 
 type Package struct {
-	ID         string   `json:",omitempty"`
-	Name       string   `json:",omitempty"`
-	Version    string   `json:",omitempty"`
-	Release    string   `json:",omitempty"`
-	Epoch      int      `json:",omitempty"`
-	Arch       string   `json:",omitempty"`
-	Dev        bool     `json:",omitempty"`
-	SrcName    string   `json:",omitempty"`
-	SrcVersion string   `json:",omitempty"`
-	SrcRelease string   `json:",omitempty"`
-	SrcEpoch   int      `json:",omitempty"`
-	Licenses   []string `json:",omitempty"`
-	Maintainer string   `json:",omitempty"`
+	ID         string         `json:",omitempty"`
+	Name       string         `json:",omitempty"`
+	Identifier *PkgIdentifier `json:",omitempty"`
```

Comment:
No. To be honest, I don't have a strong opinion about this. In terms of performance it depends on the size of the structs, these two reads can add more light on this topic:

- https://medium.com/a-journey-with-go/go-should-i-use-a-pointer-instead-of-a-copy-of-my-struct-44b43b104963
- https://stackoverflow.com/questions/24452323/whats-the-difference-between-pointer-and-value-in-struct

I usually prefer pointers because it's easier to determine if something is set or not: it's easier to do `foo != nil` than `foo != customType{}`

Reply:
It looks like we can just create a function `func(i PkgIdentifier) Empty() bool`.

Reply:
we don't need to change purl/cpe.
I'm worried this might get confusing
Because `Layer` and `Digest` don't use pointers.
I think we should either use pointers for each structural field (Layer, Digest, PkgIdentifier) or not use them at all.
In this case I am for the second option

Reply:
done at https://github.com/aquasecurity/trivy/pull/5439/commits/add17e817d2f79c6bb0f0543e43a03f89e165c40